AEPIdentity/Sources/IdentityState.swift

```diff
@@ -283,6 +283,30 @@ class IdentityState {
         }
     }
 
+    /// Clears all identities and regenerates a new ECID value.
+    /// Saves identities to persistence and creates a new shared state.
+    /// - Parameters:
+    ///   - event: event which triggered the reset call
+    ///   - createSharedState: function which creates new shared states
+    func resetIdentifiers(event: Event,
+                          createSharedState: ([String: Any], Event) -> Void) {
+        guard identityProperties.privacyStatus != .optedOut else { return }
+        // clear the properties
+        identityProperties.ecid = nil
+        identityProperties.advertisingIdentifier = nil
+        identityProperties.blob = nil
+        identityProperties.locationHint = nil
+        identityProperties.customerIds?.removeAll()
+        identityProperties.isAidSynced = false
+        identityProperties.pushIdentifier = nil
+        pushIdManager.updatePushId(pushId: nil)
```

Comment:
This will generate a push optout hit which doesn't seem to be required since the old ECID will no longer be used for interactions, did you think otherwise when you included it?

Reply:
DIdn't mean to include it. removed.

Reply:
Calling updatePushId will dispatch a push enabled/disabled event to Analytics. If you don't call `updatePushId`, the `pushEnabled` persisted value isn't updated to false. It appears, if I'm reading it correctly, if push was enabled and we simply clear the push ID without updating the push manager, new push enabled events won't get sent to Analytics as `pushEnabled` is already true.
If this logic is needed when privacy is set to opt-out, where the old ECID is also no longer used, I don't see why we wouldn't do it for reset identities.

Reply:
@kevinlind The original reason I had it was because we had that same logic for the opt-out so I assumed it applied as the two processes seem very similar. @emdobrin how does handle identities differentiate from the opt-out case here?